Library/Homebrew/dev-cmd/bump.rb

```diff
@@ -227,7 +237,7 @@ def retrieve_and_display_info(formula_or_cask, name, repositories, args:, ambigu
     pull_requests = if !args.no_pull_requests? && (args.named.present? ||
                        (livecheck_latest.is_a?(Version) && livecheck_latest != current_version))
```

Comment:
Looking through this again, I overlooked that we're also checking for a newer livecheck version here. This should be updated to use `>` for comparison and also check if the repology version is newer.

Since we're doing the same version checks to define `new_version`, we can define it before this and simply use `(args.named.present? || new_version)` here. I can't suggest all the changes but it looks like this:

```patch
diff --git a/Library/Homebrew/dev-cmd/bump.rb b/Library/Homebrew/dev-cmd/bump.rb
index 573cde9accc43c18142741d50d333358bdf2df11..812079771ec444cbf0d2e59fbbc91bdc31fcd87d 100644
--- a/Library/Homebrew/dev-cmd/bump.rb
+++ b/Library/Homebrew/dev-cmd/bump.rb
@@ -227,15 +227,21 @@ module Homebrew
       version_name = "cask version   "
     end
 
+    livecheck_latest = livecheck_result(formula_or_cask)
+
     repology_latest = if repositories.present?
       Repology.latest_version(repositories)
     else
       "not found"
     end
 
-    livecheck_latest = livecheck_result(formula_or_cask)
-    pull_requests = if !args.no_pull_requests? && (args.named.present? ||
-                       (livecheck_latest.is_a?(Version) && livecheck_latest != current_version))
+    new_version = if livecheck_latest.is_a?(Version) && livecheck_latest > current_version
+      livecheck_latest
+    elsif repology_latest.is_a?(Version) && repology_latest > current_version
+      repology_latest
+    end.presence
+
+    pull_requests = if !args.no_pull_requests? && (args.named.present? || new_version)
       retrieve_pull_requests(formula_or_cask, name)
     end.presence
 
@@ -256,14 +262,8 @@ module Homebrew
     EOS
 
     return unless args.open_pr?
-    return if pull_requests
-
-    new_version = if livecheck_latest.is_a?(Version) && livecheck_latest > current_version
-      livecheck_latest
-    elsif repology_latest.is_a?(Version) && repology_latest > current_version
-      repology_latest
-    end.presence
     return unless new_version
+    return if pull_requests
 
     system HOMEBREW_BREW_FILE, "bump-#{type}-pr", "--no-browse",
            "--message=Created by `brew bump`", "--version=#{new_version}"

```

The only other change in this patch is moving the `livecheck_latest` definition above `repology_latest`, for the sake of consistency (as that's how the variables are used in subsequent lines).

Reply:
@samford Thanks again, updated!